estate/models/estate_property_type.py

```diff
@@ -0,0 +1,23 @@
+from odoo import fields, models, api
+
+
+class EstatePropertyType(models.Model):
+    _name = "estate.property.type"
+    _description = "Type of real estate property"
+    _order = "sequence"
+
+    name = fields.Char(required=True, string="Type")
+    property_ids = fields.One2many("estate.property", 'property_type_id')
+    sequence = fields.Integer()
+    offer_ids = fields.One2many("estate.property.offer", "property_type_id")
+
+    @api.depends("offer_ids")
+    def _compute_offer_count(self):
+        for offer_type in self:
+            offer_type.offer_count = len(offer_type.offer_ids)
+
+    offer_count = fields.Integer(compute=_compute_offer_count)
+
+    _sql_constraints = [
```

Comment:
Order of the class

1: Class attributes (_name, _order, ..)
2: Fields
3: Constrains
4: Functions